server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/CellIdSource.java

```diff
@@ -33,11 +33,11 @@
  * to numeric long values for bucketing.
  */
 public class CellIdSource extends ValuesSource.Numeric {
-    private final ValuesSource.GeoPoint valuesSource;
+    private final ValuesSource.Geo valuesSource;
     private final int precision;
-    private final GeoPointLongEncoder encoder;
+    private final GeoGridTiler encoder;
 
-    public CellIdSource(GeoPoint valuesSource, int precision, GeoPointLongEncoder encoder) {
+    public CellIdSource(Geo valuesSource, int precision, GeoGridTiler encoder) {
```

Comment:
Unrelated to this PR, but I think we should brainstorm and try to come up with a better name for `Geo`.

Reply:
I tend to agree. More investigation into how to expand aggregation support to shape fields is necessary. If shape fields supported all the same aggregations as geo-shape fields, I believe this would be straightforward and renaming to `SPATIAL` makes sense. The problem is that things like `geo_distance` and `geo*_grid` are only relevant to geo-shape fields.